network/openssl_cert.py

```diff
@@ -0,0 +1,315 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2016, Yanis Guenane <yanis+ansible@guenane.org>
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from ansible.module_utils.basic import *
+from OpenSSL import crypto
+from random import randint
+
+import os
+
+
+DOCUMENTATION = '''
+---
+module: openssl_cert
+author: "Yanis Guenane (@Spredzy)"
+version_added: "2.2"
+short_description: Generate OpenSSL Certificates
+description:
+    - "This module allows one to (re)generates OpenSSL certificates. It implements a notion
+       of provider (ie. 'self-signed', 'letsencrypt'), it can be further extended.
+       It uses the pyOpenSSL python library to interact with openssl."
+requirements:
+    - "python-pyOpenSSL"
+options:
+    state:
+        required: false
+        default: "present"
+        choices: [ present, absent ]
+        description:
+            - Whether the certificate should exist or not, , taking action if the state is different from what is stated.
+    name:
+        required: true
+        description:
+            - Name of the generated OpenSSL certificate
```

Comment:
instead of separate name and path options, how about simplifying by combining the two into a single option that takes in the absolute path to where the generated certificate will be created:

````
  dest:
    description:
      - Remote absolute path where the generated certificate file should be created. If the parent directories do not exist, they are automatically created as long as the user has sufficient privileges.
    required: true
    default: null
    aliases: [ 'path', 'destfile' ]
````